pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/generator/WhenDirective.java

```diff
@@ -81,12 +84,29 @@ public String getDisplayName() {
         public String toGroovy(@Nonnull WhenDirective directive) {
             if (directive.conditional != null) {
                 UninstantiatedDescribable ud = UninstantiatedDescribable.from(directive.conditional);
-                DescribableModel model = ud.getModel();
+                DescribableModel<? extends DeclarativeStageConditional> model = ud.getModel();
                 if (model != null) {
+                    Map<String,Object> args = new HashMap<>();
+                    // Hack to null out non-required fields for UI prettiness.
+                    for (String argKey : ud.getArguments().keySet()) {
+                        DescribableParameter p = model.getParameter(argKey);
+                        if (p == null ||
+                                p.isRequired() ||
+                                !(String.class.equals(p.getErasedType())) ||
+                                !ud.getArguments().get(argKey).equals("")) {
+                            args.put(argKey, ud.getArguments().get(argKey));
```

Comment:
Quite inefficient, you should iterate over `ud.getArguments().entrySet()` instead of doing map lookups all the time.

Reply:
So you merged without addressing this?